RUN3/AliAnalysisTaskAO2Dconverter.cxx

```diff
@@ -45,88 +47,143 @@ ULong64_t GetEventIdAsLong(AliVHeader *header)
+    Tracks->Branch("fCYY", &fCYY, "fCYY/F");
+    Tracks->Branch("fCZY", &fCZY, "fCZY/F");
+    Tracks->Branch("fCZZ", &fCZZ, "fCZZ/F");
+    Tracks->Branch("fCSnpY", &fCSnpY, "fCSnpY/F");
+    Tracks->Branch("fCSnpZ", &fCSnpZ, "fCSnpZ/F");
+    Tracks->Branch("fCSnpSnp", &fCSnpSnp, "fCSnpSnp/F");
+    Tracks->Branch("fCTglY", &fCTglY, "fCTglY/F");
+    Tracks->Branch("fCTglZ", &fCTglZ, "fCTglZ/F");
+    Tracks->Branch("fCTglSnp", &fCTglSnp, "fCTglSnp/F");
+    Tracks->Branch("fCTglTgl", &fCTglTgl, "fCTglTgl/F");
+    Tracks->Branch("fC1PtY", &fC1PtY, "fC1PtY/F");
+    Tracks->Branch("fC1PtZ", &fC1PtZ, "fC1PtZ/F");
+    Tracks->Branch("fC1PtSnp", &fC1PtSnp, "fC1PtSnp/F");
+    Tracks->Branch("fC1PtTgl", &fC1PtTgl, "fC1PtTgl/F");
+    Tracks->Branch("fC1Pt21Pt2", &fC1Pt21Pt2, "fC1Pt21Pt2/F");
+    Tracks->Branch("fTPCinnerP", &fTPCinnerP, "fTPCinnerP/F");
+    Tracks->Branch("fFlags", &fFlags, "fFlags/l");
+    Tracks->Branch("fITSClusterMap", &fITSClusterMap, "fITSClusterMap/b");
+    Tracks->Branch("fTPCncls", &fTPCncls, "fTPCncls/s");
+    Tracks->Branch("fTRDntracklets", &fTRDntracklets, "fTRDntracklets/b");
+    Tracks->Branch("fITSchi2Ncl", &fITSchi2Ncl, "fITSchi2Ncl/F");
+    Tracks->Branch("fTPCchi2Ncl", &fTPCchi2Ncl, "fTPCchi2Ncl/F");
+    Tracks->Branch("fTRDchi2", &fTRDchi2, "fTRDchi2/F");
+    Tracks->Branch("fTOFchi2", &fTOFchi2, "fTOFchi2/F");
+    Tracks->Branch("fTPCsignal", &fTPCsignal, "fTPCsignal/F");
+    Tracks->Branch("fTRDsignal", &fTRDsignal, "fTRDsignal/F");
+    Tracks->Branch("fTOFsignal", &fTOFsignal, "fTOFsignal/F");
+    Tracks->Branch("fLength", &fLength, "fLength/F");
+  }
+  PostData(2, Tracks);
+
+  // Associate branches for Calo
+  TTree* Calo = CreateTree(kCalo);
+  Calo->SetAutoFlush(fNumberOfEventsPerCluster);
+  if (fTreeStatus[kCalo]) {
+    Calo->Branch("fEventId", &fEventId, "fEventId/l"); // same
+    Calo->Branch("fCellNumber", &fCellNumber, "fCellNumber/S");
+    Calo->Branch("fAmplitude", &fAmplitude, "fAmplitude/F");
+    Calo->Branch("fTime", &fTime, "fTime/F");
+    Calo->Branch("fType", &fType, "fType/B");
+  }
+  PostData(3, Calo);
+
+  // Associate branches for TOF
+  TTree* TOF = CreateTree(kTOF);
+  TOF->SetAutoFlush(fNumberOfEventsPerCluster);
+  if (fTreeStatus[kTOF]) {
+    TOF->Branch("fEventId", &fEventId, "fEventId/l"); // same
+    TOF->Branch("fTOFChannel", &fTOFChannel, "fTOFChannel/I");
+    TOF->Branch("fTOFncls", &fTOFncls, "fTOFncls/S");
+    TOF->Branch("fDx", &fDx, "fDx/F");
+    TOF->Branch("fDz", &fDz, "fDz/F");
+    TOF->Branch("fToT", &fToT, "fToT/F");
+  }
+  PostData(4, TOF);
+
+  Prune(); //Removing all unwanted branches (if any)
+}
+
+void AliAnalysisTaskAO2Dconverter::Prune()
+{
+  if (fPruneList.IsNull() || fPruneList.IsWhitespace())
+    return;
+  TObjArray* arr = fPruneList.Tokenize(" ");
+  for (Int_t i = 0; i < arr->GetEntries(); i++)
+    for (Int_t j = 0; j < kTrees; j++)
+      fTree[j]->SetBranchStatus(arr->At(i)->GetName(), 0);
```

Comment:
This will print out a lot of errors: for each tree that does not have the branch you will get an error message.
Do I get this correctly? Either we have a Prune list per tree or we should figure this out.